storage/database/leveldb_database.go

```diff
@@ -78,13 +80,9 @@ func GetOpenFilesLimit() int {
 	if err != nil {
 		logger.Crit("Failed to retrieve file descriptor allowance", "err", err)
 	}
-	if limit < 2048 {
-		if err := fdlimit.Raise(2048); err != nil {
-			logger.Crit("Failed to raise file descriptor allowance", "err", err)
-		}
-	}
-	if limit > 2048 { // cap database file descriptors even if more is available
-		limit = 2048
+	if limit < minFileDescriptorsForDBManager {
+		logger.Crit("Failed to raise file descriptor allowance to the minimum value",
```

Comment:
```suggestion
		logger.Crit("Raised number of file descriptor  is below the minimum value",
```

Reply:
> 1. Shouldn't we leave some fd for network and other things?

As it returns `limit / 2`, not `limit`, half of our fds are left for network and other things.

> 2. There is something strange about file limit. 19% is assigned for state trie DB. However, when we use shardedDB, each shard use 19% of total fd. Therefore, 4 shards uses 67% of fds. Could you check?

Codes which divides the given fds were missing. `copiedDBC.OpenFilesLimit /= int(numShards)` should be added after copying the original database configuration. I'll update this to this PR. Thanks!

```
	for i := 0; i < int(numShards); i++ {
		copiedDBC := *dbc
		copiedDBC.Dir = path.Join(copiedDBC.Dir, strconv.Itoa(i))
		copiedDBC.LevelDBCacheSize /= int(numShards)
		copiedDBC.OpenFilesLimit /= int(numShards) << missing part

		db, err := newDatabase(&copiedDBC, et)
		if err != nil {
			return nil, err
		}
		shards = append(shards, db)
		go batchWriteWorker(pdbBatchTaskCh)
	}
```

Reply:
Did you applied the second one? I cannot see it.

Reply:
@winnie-byun I've updated the PR, please check 😄